### PR TITLE
[Kotlin Migration] FlashCardsContract

### PR DESCRIPTION
**Pull request by n1snt**
## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented on your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by Arthur-Milchior — COMMENTED** [review]
I'm quite sorry. There is at least two issues.
First, a lot of documentation that is not ideal because it lost its identation
Second, the first commit's unit test seems not to compile.

Admittedly, it's independent of your change, it seems to have been broken for quite some time and not even tested by our CI, which is entirely another issue. But since I don't want to merge this until I know API's unit test work and API's compile in the intermediary commit, I fear this commit will remain stuck for some time

File: api/src/main/java/com/ichi2/anki/FlashCardsContract.kt [hunk 1]
```diff
@@ -4,520 +4,547 @@
...
+     *
+     *
+     *
+     *
      * Example for querying notes with a certain tag:
      * <pre>
-     *     <code>
-     *         final Cursor cursor = cr.query(FlashCardsContract.Note.CONTENT_URI,
-     *                                        null,         // projection
-     *                                        "tag:my_tag", // example query
```
[the remark this answers is not in the document]
Comment (n1snt) [reply api/src/main/java/com/ichi2/anki/FlashCardsContract.kt]: Sorry for the late reply, I'm caught up in a lot of things at the moment.
Will make this changes as soon as I get some time free.

[the remark this answers is not in the document]
Comment (Arthur-Milchior) [reply api/src/main/java/com/ichi2/anki/FlashCardsContract.kt]: No problem.
I mean it. You have absolutely no obligation towards AnkiDroid. Nobody has any obligation towards it, with the only exception of Google Summer of Code where mentor and mentees must follow GSoC rules.

If waiting for this PR were blocking anything, someone would just take over.
The worst risk if you wait a month before restarting is that somebody modified this file and you must update your PR to take their change into account. 

So, really, takes your time, come back when you want and can.

**Comment by david-allison** [comment]
Superseded by https://github.com/ankidroid/Anki-Android/pull/12232

**Pull request closed without merging**